rasa/core/channels/botframework.py

```diff
@@ -47,37 +47,36 @@ def __init__(
         self.bot = bot
 
     async def _get_headers(self) -> Optional[Dict[Text, Any]]:
-        if BotFramework.token_expiration_date < datetime.datetime.now():
-            uri = f"{MICROSOFT_OAUTH2_URL}/{MICROSOFT_OAUTH2_PATH}"
-            grant_type = "client_credentials"
-            scope = "https://api.botframework.com/.default"
-            payload = {
-                "client_id": self.app_id,
-                "client_secret": self.app_password,
-                "grant_type": grant_type,
-                "scope": scope,
-            }
+        if BotFramework.token_expiration_date >= datetime.datetime.now():
+            return BotFramework.headers
+        uri = f"{MICROSOFT_OAUTH2_URL}/{MICROSOFT_OAUTH2_PATH}"
+        grant_type = "client_credentials"
+        scope = "https://api.botframework.com/.default"
+        payload = {
+            "client_id": self.app_id,
+            "client_secret": self.app_password,
+            "grant_type": grant_type,
+            "scope": scope,
+        }
 
-            token_response = requests.post(uri, data=payload)
+        token_response = requests.post(uri, data=payload)
 
-            if token_response.ok:
-                token_data = token_response.json()
-                access_token = token_data["access_token"]
-                token_expiration = token_data["expires_in"]
+        if token_response.ok:
+            token_data = token_response.json()
+            access_token = token_data["access_token"]
+            token_expiration = token_data["expires_in"]
 
-                delta = datetime.timedelta(seconds=int(token_expiration))
-                BotFramework.token_expiration_date = datetime.datetime.now() + delta
+            delta = datetime.timedelta(seconds=int(token_expiration))
+            BotFramework.token_expiration_date = datetime.datetime.now() + delta
 
-                BotFramework.headers = {
-                    "content-type": "application/json",
-                    "Authorization": "Bearer %s" % access_token,
-                }
-                return BotFramework.headers
-            else:
-                logger.error("Could not get BotFramework token")
-                return None
-        else:
+            BotFramework.headers = {
+                "content-type": "application/json",
+                "Authorization": "Bearer %s" % access_token,
+            }
             return BotFramework.headers
+        else:
+            logger.error("Could not get BotFramework token")
+            return None
```

Comment:
Function `BotFramework._get_headers` refactored with the following changes:

- Add guard clause ([last-if-guard](https://docs.sourcery.ai/refactorings/last-if-guard/))